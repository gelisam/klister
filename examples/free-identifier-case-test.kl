#lang "n-ary-app.kl"

(import (shift "n-ary-app.kl" 1))
(import (shift "free-identifier-case.kl" 1))

(define-macros
  ([test
    (lambda (stx)
     (syntax-case stx
       [(list (_ x))
        (free-identifier-case x
          [if (pure '1)]
          [else (pure '2)]
          [define (pure '3)]
          [#%app (pure '4)]
          [(else y) (pure y)])]))]))

(define hello 32)
(example (test if))
(example (test else))
(example (test define))
(example (test #%app))
(example (test hello))
