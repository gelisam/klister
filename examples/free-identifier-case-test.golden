1 : Integer
2 : Integer
3 : Integer
4 : Integer
32 : Integer
